Atomic/sign.py

```diff
@@ -0,0 +1,107 @@
+from . import util
+from . import Atomic
+import os
+import tempfile
+from .atomic import AtomicError
+import re
+
+
+class Sign(Atomic):
+    def sign(self):
+        # TODO
+        # Atomic is run as sudo. Should we work around that?
+
+        if self.args.debug:
+            util.write_out(str(self.args))
+
+        # Verify all the images provided are valid
+        self.check_input_validity()
+
+        signature_path = util.get_signature_path(self.atomic_config)
+        use_def_sig_path = True if signature_path == self.args.output else False
```

Comment:
I see the issue is the way signature_path is being calculated. Use a single var, `signature_path`. If self.args.output overrides default, then update signature_path. Default should be in one place (you have it defined in two places). I think that means get rid of use_def_sig_path boolean and calculate the path in one place.